### ISSUE TITLE
Fields with Delimiter

### ISSUE THREAD

**Issue by wahmedswl** [james-e-morris/csv-logger#3]
Hi! Thanks for this package. CSV is not written properly if there is Delimiter in the field value. CSV package is not used in the write path.

Thanks

**Comment by james-e-morris** [comment]
Please provide a pull request on how you would fix the issue with a delimiter in the field. As with any delimited file type, if that delimiter is used within the data itself, then there will be issues.  Are you suggesting I force double quotes around the header field if the delimiter is present in the field?

This package uses Logger functionality rather than the CSV module so that the file operations act like a rotating logger to make saving data to CSV easier without needing to worry about file rollover and such. The delimiter is an input for this reason, so you can use non-comma if you want to.

**Comment by wahmedswl** [comment]
@Morrious Yes, there can be option to double quote by-default because CSV support those. CSV module is used in the code, can it be used in write as well?
https://github.com/Morrious/csv-logger/blob/b8b29c11b3b9f34578d82aa034d41e2fe58409e6/src/csv_logger/csv_logger.py#L184

**Comment by james-e-morris** [comment]
> @Morrious Yes, there can be option to double quote by-default because CSV support those. CSV module is used in the code, can it be used in write as well?
> 

Okay, that's a good idea, I'll work on that.

I use the CSV module for the read operation to make it easier for you to access your data after it's been logged, but the write operations aren't actually handled by me. This module extends the logging.Logger class, which provide the '.info()' functions and similar. What this does is just overwrite the formatting functions used by logging.Formatter. So what I'm thinking is adding in an option for double quotes for the header and for the values when being formatted. Maybe even a warning to the console of the data has commas and the double-quote bool is not set.

**Comment by wahmedswl** [comment]
Also, there was ```extras``` parameter in the ```info``` method but that wasn't logged. Can you please post eg where we can write the key values as context and that's written.

Thanks

**Comment by james-e-morris** [comment]
Are you referring to the `extra` input to the `logging.info` function? [Docs here](https://docs.python.org/3/library/logging.html#logging.debug:~:text=fourth%20keyword%20argument%20is-,extra,-which%20can%20be%20used)

This should work the same way using csv-logger as with the logging module, since I don't touch that functionality. An input to the CsvLogger init is the `fmt` string which is where you would configure your extras just like you would in the logging docs. The default for that value is just the timestamp + message, but you can change it however you would like for your logs output.
![image](https://github.com/Morrious/csv-logger/assets/32372811/5a40f0ac-0787-466e-80dd-fe2ffb291499)
